packages/wonder-blocks-dropdown/src/components/dropdown-core.tsx

```diff
@@ -544,20 +542,43 @@ class DropdownCore extends React.Component<Props, State> {
                 );
             }
 
-            const node = ReactDOM.findDOMNode(
-                focusedItemRef.ref.current,
-            ) as HTMLElement;
-            if (node) {
-                node.focus();
-                // Keep track of the original index of the newly focused item.
-                // To be used if the set of focusable items in the menu changes
-                this.focusedOriginalIndex = focusedItemRef.originalIndex;
-
-                if (onFocus) {
-                    // Call the callback with the node that was focused.
-                    onFocus(node);
+            const focusNode = () => {
+                // No point in doing work if we're not open.
+                if (!this.props.open) {
+                    return;
                 }
-            }
+
+                const node = ReactDOM.findDOMNode(
+                    focusedItemRef.ref.current,
+                ) as HTMLElement;
+
+                // If we don't have a node and we're virtualized, we need to
+                // wait for the next animation frame to focus the item.
+                if (!node && this.shouldVirtualizeList()) {
+                    // Wait for the next animation frame to focus the item,
+                    // that way the virtualized list has time to render the
+                    // item in the DOM. We do this in a recursive way as
+                    // occasionally, one frame is not enough.
+                    this.props.schedule.animationFrame(focusNode);
+                    return;
+                }
```

Comment:
**note:** This bit is new. It basically means we retry focusing if there's no node and we're virtualized. Had to do this because one animation frame only worked 90% of the time. This way, it keeps doing it till there's a node. Short of a signal from the virtualization saying "the nodes are in the DOM now", this is the best we can do.